cmd/zfs_object_agent/zoa/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+use std::ffi::CStr;
+use std::os::raw::c_char;
+use zoa_common::init;
+
+#[no_mangle]
+pub extern "C" fn libzoa_init(socket_dir_ptr: *const c_char, log_file_ptr: *const c_char) {
+    let socket_dir = unsafe {
+        CStr::from_ptr(socket_dir_ptr)
+            .to_string_lossy()
+            .into_owned()
+    };
+    let log_file = unsafe { CStr::from_ptr(log_file_ptr).to_string_lossy().into_owned() };
+
+    let verbosity: u64 = 2;
+    let file_name: Option<&str> = Some(log_file.as_str());
```

Comment:
I don't think you need to specify the types here
```suggestion
    let verbosity = 2;
    let file_name = Some(&log_file);
```